components/code-editor/editor.js

```diff
@@ -99,12 +99,12 @@ class CodeEditor extends Component {
 		}
 
 		if ( ! this.props.focus && this.editor.hasFocus() ) {
-			document.activeElement.blur();
+			document.activeElement.change();
 		}
 	}
 
 	render() {
-		return <textarea ref={ ( ref ) => ( this.textarea = ref ) } value={ this.props.value } />;
+		return <textarea ref={ ( ref ) => ( this.textarea = ref ) } value={ this.props.value } onChange={ this.props.onChange } />;
```

Comment:
We're calling `this.props.onChange()` both here in the text field's `onChange` handler and also in our `this.editor.on( 'change', ... )` handler. I fear that this means we're calling `this.props.onChange()` twice on every keystroke. 

I think that a better approach would be to make this text area an uncontrolled component and then let CodeMirror be solely responsible for handling user input and dispatching its `change` event.

```js
return <textarea ref={ ( ref ) => ( this.textarea = ref ) } defaultValue={ this.props.value } />;

```

https://reactjs.org/docs/uncontrolled-components.html